pkg/operation/hybridbotanist/controlplane.go

```diff
@@ -337,6 +362,13 @@ func (b *HybridBotanist) DeployKubeAPIServer() error {
 		return err
 	}
 
+	// If shoot is hibernated we don't want the HPA to interfer with our scaling decisions.
+	if b.Shoot.IsHibernated {
```

Comment:
Shouldn't we do that before applying the Kube-Apiserver deployment?

Reply:
If we do that before the `Apply()` call then the HPA will be redeployed.

Reply:
As discussed, I have adapted the kube-apiserver chart to set min=max=0 for the HPA if replicas=0.